src/vs/workbench/services/runtimeStartup/common/runtimeStartup.ts

```diff
@@ -335,6 +352,8 @@ export class RuntimeStartupService extends Disposable implements IRuntimeStartup
 	 */
 	completeDiscovery(): void {
 		this._startupPhase.set(RuntimeStartupPhase.Complete, undefined);
+		// Fire the onDidCompleteDiscoveryPhase event.
+		this.onDidCompleteDiscoveryPhaseEmitter.fire(undefined);
```

Comment:
Maybe you already saw this but just in case! Thoughts on consolidating the handling here instead of adding a new event+emitter?

https://github.com/posit-dev/positron/blob/8083208f2b6924690d7ea579f866976a3a9a5f15/src/vs/workbench/services/runtimeStartup/common/runtimeStartup.ts#L165-L183

Reply:
I think you are right that I can avoid adding the new event+emitter and handle things as part of `onDidChangeRuntimeStartupPhase`. Thanks for pointing this out!